gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -291,6 +291,10 @@ private boolean addSpecHelperMethod(Spec spec) {
    */
   @VisibleForTesting
   public static boolean isWithinRange(String cronExpression, int maxNumDaysToScheduleWithin) {
+    if (cronExpression == null || cronExpression.trim().isEmpty()) {
+      // If the cron expression is empty or null, return true to capture adhoc flows
+      return true;
+    }
```

Comment:
at the core, I understand why we don't continue w/ `cronExpression == null`... but is this really happening?  more canonical would be to have callers avoid calling this w/ `null`.  then add a `Preconditions` here that croaks if `null` is passed in.